document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/mapping/ApiToDefendantMapperTest.java

```diff
@@ -0,0 +1,87 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.model.prosecution.defendant.AddressApi;
+import uk.gov.companieshouse.api.model.prosecution.defendant.CompanyOfficerDetailsApi;
+import uk.gov.companieshouse.api.model.prosecution.defendant.DefendantApi;
+import uk.gov.companieshouse.api.model.prosecution.defendant.PersonOfficerDetailsApi;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.mappers.ApiToDefendantMapper;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.mappers.ApiToDefendantMapperImpl;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Address;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.CompanyOfficerDetails;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Defendant;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.PersonOfficerDetails;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ApiToDefendantMapperTest {
+
+    private ApiToDefendantMapper apiToDefendantMapper = new ApiToDefendantMapperImpl();
+
+    private static final String OFFICER_ID = "officerId";
+    private static final AddressApi ADDRESS =
+            new AddressApi("1", "street", "area", "town", "region", "country", "postcode");
+    private static final PersonOfficerDetailsApi PERSON_OFFICER_DETAILS = new PersonOfficerDetailsApi();
+    private static final CompanyOfficerDetailsApi COMPANY_OFFICER_DETAILS = new CompanyOfficerDetailsApi();
+    private static final LocalDate DATE_APPOINTED_ON = LocalDate.now();
+    private static final LocalDate DATE_TERMINATED_ON = LocalDate.now();
+    private static final String APPOINTMENT_TYPE = "appointmentType";
+    private static final boolean IS_CORPORATE_APPOINTMENT = true;
+
+    @Test
+    @DisplayName("Tests defendant API values map to defendant DocGen model")
+    void testApiToDefendantMaps() {
+        Defendant defendant = apiToDefendantMapper.apiToDefendant(createDefendant());
+
+        assertNotNull(defendant);
+        assertEquals(OFFICER_ID, defendant.getOfficerId());
+        assertEquals(ADDRESS.getHouseNameNumber(), defendant.getAddress().getHouseNameNumber());
+        assertEquals(ADDRESS.getStreet(), defendant.getAddress().getStreet());
+        assertEquals(ADDRESS.getArea(), defendant.getAddress().getArea());
+        assertEquals(ADDRESS.getPostTown(), defendant.getAddress().getPostTown());
+        assertEquals(ADDRESS.getRegion(), defendant.getAddress().getRegion());
+        assertEquals(ADDRESS.getRegion(), defendant.getAddress().getRegion());
```

Comment:
The region assertion has been duplicated here. Country and postcode assertions are missing.

Reply:
👍 Will change.